codeclimate/systems.yaml

```
@@ -0,0 +1,87 @@
+---
+systems:
+  codeclimate:
+    description: |
+      This system enables Honeydipper to integrate with `CodeClimate`.
+
+    meta:
+      configurations:
+        - name: api_key
+          description: The token for authenticating with CodeClimate
+        - name: url
+          description: The CodeClimate API URL
+
+      notes:
+        - For example
+        - example: |
+            ---
+            systems:
+              codeclimate:
+                data:
+                  token: ENC[gcloud-kms,...masked...]
+                  url: "https://api.codeclimate.com/v1/"
+        - |
+          To configure the integration in CodeClimate,
+
+          1. navigate to :code:`User Settings` => :code:`API Access`
+          2. generate a new token, and record it as :code:`token` in system data
+
+        - |
+          .. code-block::
+
+             https://myhoneydipper.com/webhook/codeclimate?token=...masked...
+
+    data:
+      api_key: _place_holder_
+      url: https://api.codeclimate.com/v1/
+
+    functions:
+      api:
+        driver: web
+        rawAction: request
+        parameters:
+          URL: '{{ .sysData.url }}/{{ .ctx.request_path }}'
+          header:
+            Authorization: Token token=$sysData.api_key
```

```suggestion
            Authorization: Token token={{ .sysData.api_key }}
```

`$` interpolation works only when `$` is the first character. [docs here](https://honeydipper.readthedocs.io/en/latest/docs/interpolation.html#referencing-context-data-with-given-path)